satellite/orders/endpoint.go

```diff
@@ -142,32 +139,15 @@ func (endpoint *Endpoint) Settlement(stream pb.Orders_SettlementServer) (err err
 				return Error.New("unable to verify order limit")
 			}
 
-			if orderLimit.DeprecatedUplinkId == nil { // new signature handling
-				if err := signing.VerifyUplinkOrderSignature(ctx, orderLimit.UplinkPublicKey, order); err != nil {
-					return Error.New("unable to verify order")
-				}
-			} else {
-				var uplinkSignee signing.Signee
-
-				// who asked for this order: uplink (get/put/del) or satellite (get_repair/put_repair/audit)
-				if endpoint.satelliteSignee.ID() == *orderLimit.DeprecatedUplinkId {
-					uplinkSignee = endpoint.satelliteSignee
-				} else {
-					uplinkPubKey, err := endpoint.certdb.GetPublicKey(ctx, *orderLimit.DeprecatedUplinkId)
-					if err != nil {
-						log.Warn("unable to find uplink public key", zap.Error(err))
-						return status.Errorf(codes.Internal, "unable to find uplink public key")
-					}
-					uplinkSignee = &signing.PublicKey{
-						Self: *orderLimit.DeprecatedUplinkId,
-						Key:  uplinkPubKey,
-					}
-				}
-				if err := signing.VerifyOrderSignature(ctx, uplinkSignee, order); err != nil {
-					return Error.New("unable to verify order")
-				}
+			// satellite verifies that it signed the order limit
+			if err := signing.VerifyUplinkOrderSignature(ctx, orderLimit.UplinkPublicKey, order); err != nil {
+				return Error.New("unable to verify order")
 			}
 
+			// TODO: add here .. satellite verifies that the order signature matches pub key in order limit
+			// TODO: implementation logic would be read back the stored public key from DB and verifies the
+			// public key stored in the order limit. Will be implemented in next PR.  [KISHORE]
```

Comment:
It's already doing it.

Reply:
where is it doing? am bit lost ... I see **VerifyOrderLimitSignature()** & **VerifyUplinkOrderSignature()** are doing it..but it is verifying based on the public key passed in the orderlimit, which a rouge node can change it. dont we need to access the corresponding uplink id's public key from a DB and verify?

Reply:
A bad storage node cannot change the order limit, since it's signed by the satellite.

Reply:
OOO ... my bad..... got it there is a satellite signature... yep got it thnx... is it looks good? Also in that case we never need to store uplink's public key. am I right?

Reply:
We wouldn't have enough storage to keep them. So, yes, we shouldn't need them anymore.

Reply:
now it made sense... I cleaned up the code ... please let me know what you think.